apps/content-publishing-api/src/controllers/v3/content.controller.v3.ts

```diff
@@ -0,0 +1,143 @@
+  @ApiOperation({
+    summary: 'Upload files and create batch announcements',
+    description:
+      'Upload files with their corresponding schema IDs to create batch announcements. Each file must be accompanied by a schemaId field.',
+  })
+  @ApiBody({
+    description: 'Asset files',
+    type: FilesUploadDto,
+  })
+  @ApiResponse({ status: '2XX', type: BatchAnnouncementResponseDto })
+  async postUploadBatchAnnouncement(
+    @Req() req: Request,
+    @Res({ passthrough: true }) _res: Response,
+  ): Promise<BatchAnnouncementResponseDto> {
+    const busboy = Busboy({ headers: req.headers });
+
+    const fileProcessingPromises: Promise<IFileResponse>[] = [];
+    let resolveResponse: (val: BatchAnnouncementResponseDto) => void;
+    const result = new Promise<BatchAnnouncementResponseDto>((resolve) => {
+      resolveResponse = resolve;
+    });
+    let fileIndex = 0;
+    const schemaIds: number[] = [];
+
+    busboy.on('field', (fieldname, value) => {
+      if (fieldname === 'schemaId') {
+        schemaIds.push(parseInt(value, 10));
+      } else {
+        throw new BadRequestException(`Unexpected field: ${fieldname}`);
+      }
+    });
+
+    busboy.on('file', async (_fieldname, fileStream, fileinfo) => {
+      fileIndex += 1;
+      if (fileIndex > this.appConfig.fileUploadCountLimit) {
+        fileStream.resume(); // Make sure we consume the entire file stream
+        throw new BadRequestException('Max file upload count per request exceeded');
+      }
+
+      fileProcessingPromises.push(
+        this.apiService.uploadStreamedAsset(fileStream, fileinfo.filename, fileinfo.mimeType),
+      );
+    });
+
+    busboy.on('finish', async () => {
+      try {
+        if (fileIndex === 0) {
+          throw new BadRequestException('No files provided in the request');
+        }
+
+        // Validate schema IDs match file count
+        if (schemaIds.length !== fileProcessingPromises.length) {
+          throw new BadRequestException('Number of schema IDs does not match number of files');
+        }
+
+        // Process uploaded files
+        const uploadResults = await Promise.all(fileProcessingPromises);
+
+        // Check for upload errors
+        const errors = uploadResults.filter((uploadResult) => uploadResult.error);
+        if (errors.length > 0) {
+          throw new BadRequestException(errors[0].error || 'File upload failed');
+        }
+
+        // Process each file and schema ID pair
+        const batchPromises = uploadResults.map((uploadResult) => {
+          if (!uploadResult.cid) {
+            return null;
+          }
+          return this.apiService
+            .enqueueBatchRequest({
+              cid: uploadResult.cid,
+              schemaId: schemaIds[uploadResults.indexOf(uploadResult)],
+            })
+            .catch(() => null);
+        });
+
+        // Submit all batch requests and collect responses
+        const batchResults = await Promise.all(batchPromises);
```

Comment:
This `Promise.all` is okay because we map rejections to `null` with a `.catch(() => null)` above